posts/MichelaRocchettiBlogPost/Michela_JSoC.qmd

```diff
@@ -0,0 +1,111 @@
+    updating already present function, determine if updating is needed and test them
+    - Develop functionality similar to the get_metadata_nghis function present in ipumsr
+
+2. Update IPUMS documentation
+    - Set-up and deploy of DocumenterVitepress.jl  
+    - Write a blog post on how IPUMS.jl can compose within the ecosystem.
+
+
+# Workflow 
+The first task was to migrate documents from Documenter to DocumenterVitepress.The aim of this issue is to assist with the significant refactor underway across JuliaHealth, 
+aimed at enhancing the discoverability and cohesion of the JuliaHealth ecosystem, particularly in relation to documentation.This issue is intended to offer guidance on migrating
+the IPUMS.jl documentation site from Documenter.jl to DocumenterVitepress.jl, in order to create a more engaging entry point for new Julia users interested in health research within the Julia community.
+To complete this task, what was done was to add a dependency of DocumenterVitepress into the docs directory of the IPUMS.jl repository. 
+When this was completed, the file Documenter.jl make.jl were migrated into a DocumenterVitepress.jl make.jl file. Working on the make.jl file pages, paragraphs and subparagraphs were added, in order to have
+a structure for the webpage explaining the IPUMS.jl package. In this prespective, some pages were added:
+  1. Home: to explain the main purpose of this work and to give general informations.
+  2. Workflows: to explain the work process behind.
+  3. How to: to give general information on how to make things work.
+  4. Tutorials: some tutorials on what is possible to do with IPUMS.jl and how to make it work 
+  5. Examples: some activity examples.
+  6. Mission: to better explain why IPUMS.jl package is useful for the community and its final objective and possible implications
+  7. References: references used to write the pages down.
+
+This first task requires some time, in particular to set up github  and cloning the repository locally in a correct manner.
+
+The second task consisted in updating the documentation across IPUMS.jl.To do that, function within the model folder in the IPUMS.jl folder were modified. The main objective of this task was to provide
+a description of the function and its attributes and an example of possible implementation and outcome. The documentation to be updated was of various type of function:
+  1. DataExtract
+  2. Dataset
+  3. DataTable
+  4. TimeSeriesTable
+  5. Error
+  6. Shapefile.
+  Each of those macrocategories (from 1 to 4) contains a set of function each, signaling the  different expected output and specific purpose.
+  Information about what each function does and about the meaning of each specific input variable, were found on the IPUMS website and references were given in writing down
+  the documentation. 
+  After writing down the description of function and inputs, examples were then formulated starting from IPUMS website: signing up to ipums, an API key 
```

Comment:
Could you include a link to this sign-up?

Reply:
I actually really like the outline you have here! Could you also write more about how to set-up working with the API? Maybe as a list like:

1. Create an account
2. Fill out ... 
3. .... 

Something like that.